tests/phpunit/tests/query/cacheResults.php

```diff
@@ -0,0 +1,707 @@
+				'args' => array(
+					'cache_results' => true,
+					'nopaging'      => true,
+				),
+			),
+			'cache true and post type any'                => array(
+				'args' => array(
+					'cache_results' => true,
+					'nopaging'      => true,
+					'post_type'     => 'any',
+				),
+			),
+			'cache true and get all'                      => array(
+				'args' => array(
+					'cache_results'  => true,
+					'fields'         => 'all',
+					'posts_per_page' => -1,
+					'post_status'    => 'any',
+					'post_type'      => 'any',
+				),
+			),
+			'cache true and page'                         => array(
+				'args' => array(
+					'cache_results' => true,
+					'post_type'     => 'page',
+				),
+			),
+			'cache true and ids'                          => array(
+				'args' => array(
+					'cache_results' => true,
+					'fields'        => 'ids',
+				),
+			),
+			'cache true and id=>parent and no found rows' => array(
+				'args' => array(
+					'cache_results' => true,
+					'fields'        => 'id=>parent',
+				),
+			),
+			'cache true and ids and no found rows'        => array(
+				'args' => array(
+					'no_found_rows' => true,
+					'cache_results' => true,
+					'fields'        => 'ids',
+				),
+			),
+			'cache true and id=>parent'                   => array(
+				'args' => array(
+					'no_found_rows' => true,
+					'cache_results' => true,
+					'fields'        => 'id=>parent',
+				),
+			),
+			'cache and ignore_sticky_posts'               => array(
+				'args' => array(
+					'cache_results'       => true,
+					'ignore_sticky_posts' => true,
+				),
+			),
+			'cache meta query'                            => array(
+				'args' => array(
+					'cache_results' => true,
+					'meta_query'    => array(
+						array(
+							'key' => 'color',
+						),
+					),
+				),
+			),
+			'cache comment_count'                         => array(
+				'args' => array(
+					'cache_results' => true,
+					'comment_count' => 0,
+				),
+			),
+			'cache term query'                            => array(
+				'args' => array(
+					'cache_results' => true,
+					'tax_query'     => array(
+						array(
+							'taxonomy' => 'category',
+							'terms'    => array( 'foo' ),
+							'field'    => 'slug',
+						),
+					),
+				),
+			),
+		);
+	}
+
+	/**
+	 * @ticket 22176
+	 */
+	public function test_query_cache_filter_request() {
+		$args   = array(
+			'cache_results' => true,
+			'fields'        => 'ids',
+		);
+		$query1 = new WP_Query();
+		$query1->query( $args );
+		$queries_before = get_num_queries();
+
+		add_filter( 'posts_request', array( $this, 'filter_posts_request' ) );
+
+		$query2 = new WP_Query();
+		$query2->query( $args );
+
+		remove_filter( 'posts_request', array( $this, 'filter_posts_request' ) );
+
+		$queries_after = get_num_queries();
+
+		$this->assertNotSame( $queries_before, $queries_after );
+	}
+
+	/**
+	 * @ticket 22176
+	 */
+	public function test_query_cache_filter_no_caching() {
```

Comment:
```suggestion
	public function test_query_cache_filter_no_caching() {
```

The name suggests a filter ought to be added within this test.